model/riscv_insts_zvkned.sail

```diff
@@ -0,0 +1,342 @@
+
+mapping clause encdec = ZVKVAESDMTYPE(ZVK_VAESDMVS, vs2, vd)
+  <-> 0b1010011 @ encdec_vreg(vs2) @ 0b00000 @ 0b010 @ encdec_vreg(vd) @ 0b1110111
+  when currentlyEnabled(Ext_Zvkned) & get_sew() == 32 & zvk_check_encdec(128, 4) & zvk_valid_reg_overlap(vs2, vd, get_lmul_pow())
+
+function clause execute (ZVKVAESDMTYPE(funct6, vs2, vd)) = {
+  let SEW      = get_sew();
+  let LMUL_pow = get_lmul_pow();
+  let num_elem = get_num_elem(LMUL_pow, SEW);
+
+  assert(SEW == 32);
+
+  let vs2_val = read_vreg(num_elem, SEW, LMUL_pow, vs2);
+  let vd_val  = read_vreg(num_elem, SEW, LMUL_pow, vd);
+
+  let eg_len   = (unsigned(vl) / 4);
+  let eg_start = (unsigned(vstart) / 4);
+
+  foreach (i from eg_start to (eg_len - 1)) {
+    assert(i * 4 + 3 < num_elem);
+
+    let state = get_velem_quad(vd_val, i);
+    let rkey  = if funct6 == ZVK_VAESDMVV
+                then get_velem_quad(vs2_val, i)
+                else get_velem_quad(vs2_val, 0);
+    let sr    = aes_shift_rows_inv(state);
+    let sb    = aes_subbytes_inv(sr);
+    let ark   = sb ^ rkey;
+    let mix   = aes_mixcolumns_inv(ark);
+
+    write_velem_quad(vd, SEW, mix, i);
+  };
+
+  set_vstart(zeros());
+  RETIRE_SUCCESS
+}
+
+mapping vaesdm_mnemonic : zvkfunct6 <-> string = {
+  ZVK_VAESDMVV <-> "vaesdm.vv",
+  ZVK_VAESDMVS <-> "vaesdm.vs",
+}
+
+mapping clause assembly = ZVKVAESDMTYPE(funct6, vs2, vd)
+  <-> vaesdm_mnemonic(funct6) ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2)
+
+union clause ast = ZVKVAESEFTYPE : (zvkfunct6, vregidx, vregidx)
+
+mapping clause encdec = ZVKVAESEFTYPE(ZVK_VAESEFVV, vs2, vd)
+  <-> 0b1010001 @ encdec_vreg(vs2) @ 0b00011 @ 0b010 @ encdec_vreg(vd) @ 0b1110111
+  when currentlyEnabled(Ext_Zvkned) & get_sew() == 32 & zvk_check_encdec(128, 4)
+
+mapping clause encdec = ZVKVAESEFTYPE(ZVK_VAESEFVS, vs2, vd)
+  <-> 0b1010011 @ encdec_vreg(vs2) @ 0b00011 @ 0b010 @ encdec_vreg(vd) @ 0b1110111
+  when currentlyEnabled(Ext_Zvkned) & get_sew() == 32 & zvk_check_encdec(128, 4) & zvk_valid_reg_overlap(vs2, vd, get_lmul_pow())
+
+function clause execute (ZVKVAESEFTYPE(funct6, vs2, vd)) = {
+  let SEW = get_sew();
+  let LMUL_pow = get_lmul_pow();
+  let num_elem = get_num_elem(LMUL_pow, SEW);
+
+  assert(SEW == 32);
+
+  let vs2_val = read_vreg(num_elem, SEW, LMUL_pow, vs2);
+  let vd_val  = read_vreg(num_elem, SEW, LMUL_pow, vd);
+
+  let eg_len   = (unsigned(vl) / 4);
+  let eg_start = (unsigned(vstart) / 4);
+
+  foreach (i from eg_start to (eg_len - 1)) {
+    assert(i * 4 + 3 < num_elem);
+
+    let state = get_velem_quad(vd_val, i);
+    let rkey  = if funct6 == ZVK_VAESEFVV
+                then get_velem_quad(vs2_val, i)
+                else get_velem_quad(vs2_val, 0);
+    let sb    = aes_subbytes_fwd(state);
+    let sr    = aes_shift_rows_fwd(sb);
+    let ark   = sr ^ rkey;
+
+    write_velem_quad(vd, SEW, ark, i);
+  };
+
+  set_vstart(zeros());
+  RETIRE_SUCCESS
+}
+
+mapping vaesef_mnemonic : zvkfunct6 <-> string = {
```

Comment:
Splitting up `zvkfunct6` will help make this mapping complete; right now, it isn't.